CONTRIBUTING.md

```diff
@@ -142,34 +142,36 @@ You might use the following command to fix auto-correctable issues, and report t
 
 ## Null safety
 
-The project uses [Checker Framework](https://checkerframework.org/) for verification of the null safety
-(see [PR 1814](https://github.com/pgjdbc/pgjdbc/pull/1814)).
+The project uses the [Checker Framework](https://checkerframework.org/) for verification of the null safety
+(this was introduced in [PR 1814](https://github.com/pgjdbc/pgjdbc/pull/1814)).
 
 By default, parameters, return values, and fields are `@NonNull`, so you need to add `@Nullable`
 as required.
 
-To execute checkerframework locally please use the following command:
+To execute the Checker Framework locally please use the following command:
 
     ./gradlew -PenableCheckerframework :postgresql:classes
 
 Notable items:
 
-* CheckerFramework verifies code method by method. That means, it does can't account for method execution order.
+* The Checker Framework verifies code method by method. That means, it can't account for method execution order.
 That is why `@Nullable` fields should be verified in each method where they are used.
 If you split logic into multiple methods, you might want verify null once, then pass it via non-nullable parameters.
+For fields that start as null and become non-null later, use `@MonotonicNonNull`.
+For fields that have already been checked against null, use `@RequiresNonNull`.
 
 * If you are absolutely sure the value is non-null, you might use `org.postgresql.util.internal.Nullness.castNonNull(T)`
 or `org.postgresql.util.internal.Nullness.castNonNull(T, String)`.
 
 * You can configure postfix completion in IntelliJ IDEA via `Preferences -> Editor -> General -> Postfix Completion`
 key: `cnn`, applicable element type: `non-primitive`, `org.postgresql.util.internal.Nullness.castNonNull($EXPR$)`.  
 
-* CheckerFramework comes with an annotated JDK, however, there might be invalid annotations.
+* The Checker Framework comes with an annotated JDK, however, there might be invalid annotations.
 In that cases, stub files can be placed to `/config/checkerframework` to override the annotations.
 It is important the files have `.astub` extension otherwise they will be ignored.
 
-* Type annotations are not that obvious when arrays come into the equation.
-You can find more details in [Java Language Specification](https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.7.4).
+* In array types, a type annotation appears immediately before the type component (either the array or the array component) it refers to.
```

Comment:
I'm afraid, it is not that easy :-(
Have you seen https://twitter.com/tagir_valeev/status/1301443081296842752 and https://twitter.com/tagir_valeev/status/1302173719398436864 ?

Reply:
I think there are two separate issues here:

 * Type annotations are written before the corresponding type, as a simple, logical extension of Java's syntax.  (That's what my proposed text says.)
 * Java's array declaration syntax is baroque.  I agree, and I am not making any excuses for it.

If you disregard the deprecated ability to write array brackets after an identifier, then Java's array syntax is not so bad.  Programmers should never write array brackets after an identifier in a declaration.

The Twitter thread shows that some programmers have trouble understanding the syntax when there are brackets after an identifier in a declaration.  The Twitter thread supports the long-standing fact that that syntax is deprecated and should not be used.